File: 2017/01/01.py
def captcha_one(s):
    cnt = 0
    for i in range(len(s) - 1):
        if s[i] == s[i+1]:
            cnt += int(s[i])
    if s[0] == s[-1]:
        cnt += int(s[0])
    return cnt


def captcha_two(s):
    l = len(s)
    cnt = 0
    for i in range(len(s)):
        if s[i] == s[(i + l//2) % l]:
            cnt += int(s[i])
    return cnt


with open('input') as f:
    s = f.readline().strip()


print('Part 1:', captcha_one(s))
print('Part 2:', captcha_two(s))

File: 2017/01/01.go
package main

import "fmt"
import "io/ioutil"
import "strconv"

func parse(file string) []int {
	numbers := []int{}
	data, _ := ioutil.ReadFile(file)
	for _, c := range string(data) {
		n, ok := strconv.Atoi(string(c))
		if ok == nil {
			numbers = append(numbers, n)
		}
	}
	return numbers
}

func checkSum(numbers []int, offset int) int {
	s := 0
	for i := 0; i < len(numbers); i++ {
		j := (i + offset) % len(numbers)
		if numbers[i] == numbers[j] {
			s += numbers[i]
		}
	}
	return s
}

func main() {
	numbers := parse("input")
	fmt.Println("Part 1:", checkSum(numbers, 1))
	fmt.Println("Part 2:", checkSum(numbers, int(len(numbers)/2)))
}
